tf2onnx/tfonnx.py

```diff
@@ -1720,6 +1720,34 @@ def mknode(op_type, inputs, opname, **kwargs):
     return nodes
 
 
+def sparse_softmax_cross_entropy_with_logits_op(ctx, node, name, args):
+    # make subgraph to implement one_hot, idea comes from onehot_op
+    indices_name = node.input[1]
+    indices_shape = ctx.get_shape(indices_name)
+    if len(indices_shape) != 1:
+        # TODO: this works for rank=1 but tensorflow supports more than this.
+        # Same principle should work but we need to implement our own eye.
+        raise ValueError("onehot op: only rank1 is supported")
+    logit_name = node.input[0]
+    depth = ctx.get_shape(logit_name)[-1]
+    dtype = utils.ONNX_TO_NUMPY_DTYPE[ctx.get_dtype(logit_name)]
+    eye = np.eye(depth).astype(dtype)
+    const_name = utils.make_name("const_eye")
+    const_eye = ctx.make_const(name=const_name, np_val=eye)
+    onehot = ctx.make_node(op_type="Gather", inputs=[const_eye.output[0], indices_name], attr={"axis": 0})
+    log_softmax = ctx.make_node(op_type="LogSoftmax", inputs=[logit_name])
+    # implement tf.multiply(np.float32(-1.0), tf.reduce_sum(tf.multiply(one_hot, log_softmax), axis=1))
+    mul1 = ctx.make_node(op_type="Mul", inputs=[onehot.output[0], log_softmax.output[0]])
+    reduce_sum = ctx.make_node(op_type="ReduceSum", inputs=[mul1.output[0]], attr={"axes": [1]})
+    const_name = utils.make_name("const_negative_one")
+    const_negative_one = ctx.make_const(name=const_name, np_val=np.array(-1).astype(dtype))
+    mul2 = ctx.make_node(op_type="Mul", inputs=[const_negative_one.output[0], reduce_sum.output[0]])
+    res = ctx.make_node(op_type="Squeeze", inputs=[mul2.output[0]], outputs=[node.output[0]], attr={"axes": [1]})
+
+    nodes = []
+    nodes.extend([onehot, log_softmax, mul1, reduce_sum, mul2, res])
```

Comment:
why not just: ```nodes = [onehot, log_softmax, mul1, reduce_sum, mul2, res]```

Reply:
@guschmue resolved, please help to merge it. Thanks.